contrib/net/http/context.go

```diff
@@ -0,0 +1,43 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package http
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+func Get(ctx context.Context, url string) (resp *http.Response, err error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	return http.DefaultClient.Do(req)
+}
+
+func Head(ctx context.Context, url string) (resp *http.Response, err error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodHead, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	return http.DefaultClient.Do(req)
+}
+
+func Post(ctx context.Context, url string, contentType string, body io.Reader) (resp *http.Response, err error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", contentType)
+	return http.DefaultClient.Do(req)
+}
+
+func PostForm(ctx context.Context, url string, data url.Values) (resp *http.Response, err error) {
+	return Post(ctx, url, "application/x-www-form-urlencoded", strings.NewReader(data.Encode()))
+}
```

Comment:
@RomainMuller @eliottness Could this be in a subpackage `client`? These functions are going to look a bit weird to me along the rest of the tracing API for this package:

![image](https://github.com/user-attachments/assets/4a25e1fe-4399-4d21-916c-7d73dbf5fe2c)

Reply:
Good point. Done :)